src/lib/mina_base/account.ml

```diff
@@ -580,17 +580,37 @@ module Checked = struct
     (*Note: Untimed accounts will always have zero min balance*)
     Boolean.not zero_min_balance
 
-  let has_permission ~to_ (account : var) =
+  let has_permission ?signature_verifies ~to_ (account : var) =
+    let signature_verifies =
+      match signature_verifies with
+      | Some signature_verifies ->
+          signature_verifies
+      | None -> (
+          match to_ with
+          | `Send ->
+              Boolean.true_
+          | `Receive ->
+              Boolean.false_
+          | `Set_delegate ->
+              Boolean.true_
+          | `Access ->
+              failwith
```

Comment:
why put a crasher here? seems unsafe?

Reply:
There's no sensible default value for `access`, so this ensures that we call it correctly everywhere.